[WPE] WPEPlatform: add WPEViewAccessible interface

carlosgcampos opened the pull request:
#### 439dc000a9675b6628215a3081f7c7f5f8c2c376
<pre>
[WPE] WPEPlatform: add WPEViewAccessible interface
<a href="https://bugs.webkit.org/show_bug.cgi?id=289227">https://bugs.webkit.org/show_bug.cgi?id=289227</a>

Reviewed by Adrian Perez de Castro.

To allow platforms to provide an accessible view object to bind to the
web process a11y tree.

* Source/WebKit/UIProcess/API/wpe/PageClientImpl.cpp:
(WebKit::PageClientImpl::accessible):
* Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp:
(WKWPE::View::~View):
(WKWPE::View::accessible const): Deleted.
* Source/WebKit/UIProcess/API/wpe/WPEWebView.h:
* Source/WebKit/UIProcess/API/wpe/WPEWebViewLegacy.cpp:
(WKWPE::ViewLegacy::~ViewLegacy):
(WKWPE::ViewLegacy::accessible const):
* Source/WebKit/UIProcess/API/wpe/WPEWebViewLegacy.h:
* Source/WebKit/UIProcess/wpe/WebPageProxyWPE.cpp:
(WebKit::WebPageProxy::bindAccessibilityTree):
* Source/WebKit/WPEPlatform/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
(wpe_view_set_gesture_controller):
(wpe_view_get_gesture_controller):
* Source/WebKit/WPEPlatform/wpe/WPEView.h:
* Source/WebKit/WPEPlatform/wpe/WPEViewAccessible.cpp: Added.
(wpe_view_accessible_default_init):
(wpe_view_accessible_bind):
* Source/WebKit/WPEPlatform/wpe/WPEViewAccessible.h: Added.
* Source/WebKit/WPEPlatform/wpe/wpe-platform.h:

Canonical link: <a href="https://commits.webkit.org/291762@main">https://commits.webkit.org/291762@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/409a01052b03883b766052d1e6297f44591a6fd6

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/93713 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/13286 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/3022 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/98719 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/44239 "Built successfully") 
| | [❌ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/13582 "Hash 409a0105 for PR 41988 does not build (failure)") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/21729 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/71541 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/28911 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/96715 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/130/builds/13582 "Hash 409a0105 for PR 41988 does not build (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/84697 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/51875 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/130/builds/13582 "Hash 409a0105 for PR 41988 does not build (failure)") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/43554 "Built successfully") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/130/builds/13582 "Hash 409a0105 for PR 41988 does not build (failure)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/2392 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/100753 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/20765 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/15149 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/80556 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/21017 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/80639 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/79894 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/24444 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/1795 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/13918 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15068 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/20749 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/25927 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/20436 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/23896 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/22177 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebKit/WPEPlatform/wpe/WPEViewAccessible.cpp:
```diff
@@ -0,0 +1,58 @@
+ *
+ * A #WPEView accessible interface.
+ *
+ * This interface enables implementing accessibility.
```
Sure, but *how* does one go about it? It would be good to exlain things better in the documentation. But in general WPEPlatform needs a pass adding and improving it, so I won't consider this a blocker for landing.

carlosgcampos committed:
[WPE] WPEPlatform: add WPEViewAccessible interface
https://bugs.webkit.org/show_bug.cgi?id=289227

Reviewed by Adrian Perez de Castro.

To allow platforms to provide an accessible view object to bind to the
web process a11y tree.

* Source/WebKit/UIProcess/API/wpe/PageClientImpl.cpp:
(WebKit::PageClientImpl::accessible):
* Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp:
(WKWPE::View::~View):
(WKWPE::View::accessible const): Deleted.
* Source/WebKit/UIProcess/API/wpe/WPEWebView.h:
* Source/WebKit/UIProcess/API/wpe/WPEWebViewLegacy.cpp:
(WKWPE::ViewLegacy::~ViewLegacy):
(WKWPE::ViewLegacy::accessible const):
* Source/WebKit/UIProcess/API/wpe/WPEWebViewLegacy.h:
* Source/WebKit/UIProcess/wpe/WebPageProxyWPE.cpp:
(WebKit::WebPageProxy::bindAccessibilityTree):
* Source/WebKit/WPEPlatform/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
(wpe_view_set_gesture_controller):
(wpe_view_get_gesture_controller):
* Source/WebKit/WPEPlatform/wpe/WPEView.h:
* Source/WebKit/WPEPlatform/wpe/WPEViewAccessible.cpp: Added.
(wpe_view_accessible_default_init):
(wpe_view_accessible_bind):
* Source/WebKit/WPEPlatform/wpe/WPEViewAccessible.h: Added.
* Source/WebKit/WPEPlatform/wpe/wpe-platform.h:

Canonical link: https://commits.webkit.org/291762@main
